state/account_test.go

```diff
@@ -41,35 +40,64 @@ func TestNonce(t *testing.T) {
 
 func TestEncodeDecode(t *testing.T) {
 	require := require.New(t)
-	s1 := Account{
-		accountType: 1,
-		nonce:       0x10,
-		Balance:     big.NewInt(20000000),
-		CodeHash:    []byte("testing codehash"),
-	}
-	ss, err := s1.Serialize()
-	require.NoError(err)
-	require.NotEmpty(ss)
-	require.Equal(66, len(ss))
 
-	s2 := Account{}
-	require.NoError(s2.Deserialize(ss))
-	require.Equal(s1.accountType, s2.accountType)
-	require.Equal(s1.Balance, s2.Balance)
-	require.Equal(s1.nonce, s2.nonce)
-	require.Equal(hash.ZeroHash256, s2.Root)
-	require.Equal(s1.CodeHash, s2.CodeHash)
+	for _, test := range []struct {
+		acc         Account
+		expectedLen int
+	}{
+		{
+			acc: Account{
+				accountType: 1,
+			},
+			expectedLen: 66,
+		},
+		{
+			acc: Account{
+				accountType: 0,
+			},
+			expectedLen: 64,
+		},
+	} {
+		test.acc.nonce = 0x10
+		test.acc.Balance = big.NewInt(20000000)
+		test.acc.CodeHash = []byte("testing codehash")
+		ss, err := test.acc.Serialize()
+		require.NoError(err)
+		require.NotEmpty(ss)
+		require.Equal(test.expectedLen, len(ss))
+
+		s2 := Account{}
+		require.NoError(s2.Deserialize(ss))
+		require.Equal(test.acc.accountType, s2.accountType)
+		require.Equal(test.acc.Balance, s2.Balance)
+		require.Equal(test.acc.nonce, s2.nonce)
+		require.Equal(hash.ZeroHash256, s2.Root)
+		require.Equal(test.acc.CodeHash, s2.CodeHash)
+	}
 }
 
 func TestProto(t *testing.T) {
 	require := require.New(t)
-	raw := "1201301a200000000000000000000000000000000000000000000000000000000000000000"
-	ss, _ := hex.DecodeString(raw)
-	s1 := Account{}
-	require.NoError(Deserialize(&s1, ss))
-	d, err := Serialize(s1)
-	require.NoError(err)
-	require.Equal(raw, hex.EncodeToString(d))
+
+	for _, test := range []struct {
+		acc Account
```

Comment:
same comment, `accountType: int32` is enough

Reply:
fixed